File: languages/js/src/entity/identity_v2/issuer_state_receipt.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";

export class IssuerStateReceipt {
    txHash: string;

    constructor(txHash: string) {
        this.txHash = txHash
    }

    public toProto(): identityEntitiesProto.IssuerStateReceipt {
        return identityEntitiesProto.IssuerStateReceipt.fromPartial({
            txHash: this.txHash
        });
    }

    static fromProto(
        r: identityEntitiesProto.IssuerStateReceipt
    ): IssuerStateReceipt {
        return new IssuerStateReceipt(r.txHash);
    }
}

File: languages/python/bloock/entity/identity_v2/issuer_state_receipt.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_v2_pb2 as proto

class IssuerStateReceipt:
    def __init__(self, txHash: str) -> None:
        self.txHash = txHash

    @staticmethod
    def from_proto(c: proto.IssuerStateReceipt) -> IssuerStateReceipt:
        return IssuerStateReceipt(
            txHash=c.tx_hash
        )

    def to_proto(self) -> proto.IssuerStateReceipt:
        return proto.IssuerStateReceipt(
            tx_hash=self.txHash
        )
